scripts/encode.lua

```diff
@@ -197,31 +225,84 @@ function start_encoding(from, to, settings)
     else
         output_directory = string.gsub(output_directory, "^~", os.getenv("HOME") or "~")
     end
-    local input_name = mp.get_property("filename/no-ext") or "encode"
+    local input_name
+    if youtube_video_id then
+        input_name = sanitize_filename(mp.get_property("media-title"))
+    else
+        input_name = mp.get_property("filename/no-ext") or "encode"
+    end
     local title = mp.get_property("media-title")
     local extension = get_extension(path)
     local output_name = get_output_string(output_directory, settings.output_format, input_name, extension, title, from, to, settings.profile)
     if not output_name then
         mp.osd_message("Invalid path " .. output_directory)
         return
     end
-    args[#args + 1] = utils.join_path(output_directory, output_name)
+
+    local args = {}
+
+    if youtube_video_id then
+        table.remove(input_args, 1)
+
+        local ffmpeg_input_collapsed = "ffmpeg_i:"
+        for _, val in ipairs(input_args) do
+            ffmpeg_input_collapsed = ffmpeg_input_collapsed .. val .. " "
+        end
+
+        local ffmpeg_output_collapsed = "ffmpeg_o:"
+        for _, val in ipairs(output_args) do
+            ffmpeg_output_collapsed = ffmpeg_output_collapsed .. val .. " "
+        end
+
+        args = {
+            settings.yt_dlp_command,
+            youtube_video_id,
+            "--external-downloader",
+            settings.ffmpeg_command,
+            "--external-downloader-args",
+            ffmpeg_input_collapsed:sub(1, -2),
+            "--external-downloader-args",
+            ffmpeg_output_collapsed:sub(1, -2),
+            "-o",
+            output_name,
```

Comment:
This should also be
```lua
utils.join_path(output_directory, output_name)
```
like the normal ffmpeg one.

It might also be useful to add
```lua
"--merge-output-format",
extension,
```
or it will try to use the container YouTube provided the steam in (ie. I get VP9 + WebM from YouTube, but my profile says H264 + Matroska, so the result will be H264 + WebM which fails as they are not compatible).